--- mod_auth_kerb-5.4/src/mod_auth_kerb.c~	2008-12-04 11:14:03.000000000 +0100
+++ mod_auth_kerb-5.4/src/mod_auth_kerb.c	2009-12-09 15:17:55.226027574 +0100
@@ -89,6 +89,7 @@
 #include <krb5.h>
 #ifdef HEIMDAL
 #  include <gssapi.h>
+#  include <gssapi/gssapi_krb5.h>
 #else
 #  include <gssapi/gssapi.h>
 #  include <gssapi/gssapi_generic.h>
--- mod_auth_kerb-5.4/src/mod_auth_kerb.c.orig	2026-02-28 22:50:02.049825889 +0100
+++ mod_auth_kerb-5.4/src/mod_auth_kerb.c	2026-02-28 23:03:46.340000285 +0100
@@ -1340,7 +1341,7 @@
   gss_name_t client_name = GSS_C_NO_NAME;
   gss_cred_id_t delegated_cred = GSS_C_NO_CREDENTIAL;
   OM_uint32 (KRB5_LIB_FUNCTION *accept_sec_token)
-     			 (OM_uint32 *, gss_ctx_id_t *, const gss_cred_id_t,
+			 (OM_uint32 *, gss_ctx_id_t *, gss_const_cred_id_t,
 			 const gss_buffer_t, const gss_channel_bindings_t,
 			 gss_name_t *, gss_OID *, gss_buffer_t, OM_uint32 *,
 			 OM_uint32 *, gss_cred_id_t *);
